### ISSUE TITLE
Losses also are required to be Lipschitz for regression ?

### ISSUE THREAD

**Issue by DanielWicz** [deel-ai/deel-lip#48]
I'm trying to apply your work to the regression problem. Nevertheless, I wonder if I can use any loss here or some special one. For example using MSE loss viable here ? (Since MSE is not Lipschitz for sure) or I should use other loss (e.g. MAE) ?

**Comment by cofri** [comment]
Hello @DanielWicz,

A Lipschitz network is independent from the loss: only the network `ŷ=f(x)` from inputs `x` (e.g. images) to ouputs `ŷ` (e.g. logits) is considered to be Lipschitz. You can thus use any loss between the outputs `ŷ` and the ground truth `y`, such as MSE or MAE for regression problem.

You must be aware that using Lipschitz networks for regression requires a good knowledge of your dataset, especially the Lipschitz constant of the function you try to estimate. For instance, let's say you want to estimate the square function `x²` on interval `[0:10]`. The maximum derivative on this interval is 20. So you won't be able to fit a 1-Lipschitz network on this function. A solution is to explicitly say that you want to train a 20-Lipschitz network.

The deel-lip library is mainly focused on classification: the losses provided in [losses.py](https://github.com/deel-ai/deel-lip/blob/master/deel/lip/losses.py) are aimed for classification. So they cannot be applied for regression.
Moreover in terms of model robustness, it's more difficult to express guarantees in regression, unlike classification.

If you have any other questions, feel free to ask. We'll be pleased to help you out and exchange about regression problems.
